app/services/folio_client.rb

```diff
@@ -118,20 +118,17 @@ def cancel_hold(resource, _item_key, patron_key)
   end
 
   # Cancel a hold request
-  # See https://s3.amazonaws.com/foliodocs/api/mod-patron/p/patron.html#patron_account__id__hold__holdid__cancel_post
   # @param [String] user_id the UUID of the user in FOLIO
   # @param [String] hold_id the UUID of the FOLIO hold
   def cancel_hold_request(user_id, hold_id)
-    # You would think FOLIO could look up this information and merge it itself, but no. You'd
-    # also think you could get the /circulation/requests/{id} endpoint data and use that,
-    # but the API schema is slightly different and OKAPI complains... so we have to look it
-    # up in the patron account data instead.
-    request_data = patron_account(user_id)['holds'].find { |h| h['requestId'] == hold_id }
```

Comment:
After this change, is there anywhere we're making sure the current patron actually has the hold (and didn't just input some other UUIDs into the URL params)?

Reply:
Doesn't look like we're validating this anywhere else. Do you think it's okay do something like `return unless request_data['requesterId'] == user_id` within that method? It'll blow up but that seems preferable to cancelling a request for another user, and I think is roughly equivalent to what would have happened with the previous logic. The better thing maybe would be check in the controller and return a 403 if there's a mismatch between the requesterId and the user_id, but requires some an additional api call to get the request and validate it.

Reply:
Elected to pass an empty hash in place of the request JSON to the cancel request call to FOLIO if the user id on the request doesn't match the current user.